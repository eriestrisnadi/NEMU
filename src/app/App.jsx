import React, { Component } from "react";
import { HashRouter, Route } from "react-router-dom";
import Home from "./Home";
import Run from "./Run";

class App extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const Page = {
      Home: props => {
        props = Object.assign({}, this.props, props);
        return <Home {...props} />;
      },
      Run: props => {
        props = Object.assign({}, this.props, props);
        return <Run {...props} />;
      },
    };

    return (
      <HashRouter>
        <div className="App">
          <Route exact path="/" render={Page.Home} />
          <Route exact path="/run" render={Page.Run} />
        </div>
      </HashRouter>
    );
  }
}

export default App;
