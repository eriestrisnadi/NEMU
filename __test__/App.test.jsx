import React from "react";
import App from "../src/app/App";
import renderer from "react-test-renderer";
import pkg from "../package.json";

const evt = {
  preventDefault: jest.fn(),
  dataTransfer: {
    dropEffect: "",
    files: [
      {
        name: "",
      },
    ],
  },
};

test("App should render first Home Component", () => {
  const component = renderer.create(<App title={pkg.name} />);
  let tree = component.toJSON();
  expect(tree.props.className).toEqual("App");
  expect(tree.children[0].type).toEqual("div");
  expect(tree.children[0].children[0].type).toEqual("h1");
  expect(tree.children[0].children[0].children[0]).toEqual(pkg.name);
  expect(tree).toMatchSnapshot();

  tree = tree.children[0];
  tree.props.onDragOver(evt);

  // re-rendering
  expect(component.toJSON()).toMatchSnapshot();
});
