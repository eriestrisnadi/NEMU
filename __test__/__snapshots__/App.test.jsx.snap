// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render first Home Component 1`] = `
<div
  className="App"
>
  <div
    className="uk-flex uk-flex-middle uk-flex-center Home"
    onDragOver={[Function]}
    onDrop={[Function]}
    uk-height-viewport=""
  >
    <h1>
      NEMU
    </h1>
  </div>
</div>
`;

exports[`App should render first Home Component 2`] = `
<div
  className="App"
>
  <div
    className="uk-flex uk-flex-middle uk-flex-center Home"
    onDragOver={[Function]}
    onDrop={[Function]}
    uk-height-viewport=""
  >
    <h1>
      NEMU
    </h1>
  </div>
</div>
`;
